src/main/java/com/globallogic/amcr/model/wiki/Wiki.java

```diff
@@ -0,0 +1,141 @@
+package com.globallogic.amcr.model.wiki;
+
+import com.globallogic.amcr.utils.Assert;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.UUID;
+
+@SuppressWarnings("unused")
+public class Wiki {
+
+    private UUID id;
+    @NotNull
+    private String title;
+    @NotNull
+    private String overview;
+
+    private UUID[] subImage;
+    @NotNull
+    private String subTitle;
+    @NotNull
+    private String subOverview;
+
+    private UUID diagram;
+
+    /**
+     * @param id             the id of the object
+     * @param title         the title of the wiki page
+     * @param overview          general overview of the wiki page
+     * @param subImage      image to be associated with the wiki page
+     * @param subTitle      secondary title of the wiki page
+     * @param subOverview          secondary overview of the wiki page
+     * @param diagram          diagram that speaks to the diagram component
+     */
+    public Wiki(UUID id, String title, String overview, UUID[] subImage, String subTitle, String subOverview, UUID diagram) {
+        setId(id);
+        setTitle(title);
+        setOverview(overview);
+        setSubImage(subImage);
+        setSubTitle(subTitle);
+        setSubOverview(subOverview);
+        setDiagram(diagram);
+    }
+
+
+    public Wiki() {
+    }
+
+
+
+    public UUID getId() {
+        return id;
+    }
+
+    public void setId(UUID id) {
+        this.id = Assert.assertNotNull(id, "ID cannot be null");
+    }
+
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String title) {
+        this.title = Assert.assertNotNull(title, "Title cannot be null");
+    }
+
+    public String getOverview() {
+        return overview;
+    }
+
+    public void setOverview(String overview) {
+        this.overview = Assert.assertNotNull(overview, "Overview cannot be null");
+    }
+
+
+    public UUID[] getSubImage() {
+        return subImage;
+    }
+
+    public void setSubImage(UUID[] subImage) {
+        this.subImage = Assert.assertNotNull(subImage, "Cover image cannot be null");
```

Comment:
SubImages can be null follow your model therefore you should remove the assertNotNull in the setter.

Reply:
Thanks for pointing that out